### PR TITLE
feat: add ESLint, Prettier, Husky, and lint-staged

### LINKED ISSUE #57: Add ESLint + Prettier with CI checks and git hooks
## Problem

No linter or formatter is configured. No git hooks are set up. Code style is enforced only by convention.

Relates to #28.

## Solution

1. Add ESLint with \`@typescript-eslint\` and React-specific rules
2. Add Prettier for consistent formatting
3. Add \`lint-staged\` + \`husky\` for pre-commit hooks (lint + format staged files)
4. Add a CI workflow step that runs lint + format check on every push/PR
5. Add \`pnpm lint\` and \`pnpm format\` scripts to package.json

### PR DESCRIPTION
**Pull request by ciampo**
## Summary

- Set up ESLint (flat config) with TypeScript and React plugins (`react-hooks`, `react-refresh`)
- Configure Prettier with project conventions (no semi, single quotes, trailing commas, 100 print width, LF line endings)
- Add `prettier-plugin-tailwindcss` for automatic Tailwind CSS class sorting
- Add Husky pre-commit hook running lint-staged to auto-lint/format staged files on commit
- Add GitHub Actions CI workflow (`lint.yml`) to enforce lint and format checks on every push to main and on PRs
- Harden all CI workflows with `--frozen-lockfile` and concurrency control
- Auto-format the entire existing codebase with Prettier
- Fix all ESLint errors and warnings across the codebase

### Tooling & config (new files)
- `eslint.config.js` — ESLint flat config with TypeScript-ESLint, react-hooks, react-refresh, and eslint-config-prettier
- `.prettierrc` / `.prettierignore` — Prettier config (including `prettier-plugin-tailwindcss`) and ignore patterns
- `.husky/pre-commit` — Husky v9 pre-commit hook running `pnpm exec lint-staged`
- `.github/workflows/lint.yml` — CI workflow running `pnpm lint` + `pnpm format:check`
- `package.json` — new scripts (`lint`, `lint:fix`, `format`, `format:check`, `prepare`), new devDependencies, lint-staged config

### CI hardening (all workflows)
- All 8 workflows now use `pnpm install --frozen-lockfile` to prevent accidental dependency mutations
- 5 check workflows (lint, typecheck, test-unit, test-e2e, test-visual) now have `concurrency` groups with `cancel-in-progress: true` to save CI minutes on rapid pushes
- `deploy.yml` intentionally excluded from concurrency (production deploys should not be cancelled mid-flight)

### Code changes (non-formatting)
- **`src/lib/auth-store.ts`** — Refactored from getter/setter API to an `update()` method. The `react-hooks/immutability` rule flagged direct property assignment on component props in `AuthBridge`. The new `update({ isAuthenticated, isLoading })` method is atomic and enforces the correct ordering (isAuthenticated before isLoading) at the API level. Interface properties are now `readonly`.
- **`src/router.tsx`** — `AuthBridge` calls `authStore.update(...)` instead of direct property assignment
- **`src/__tests__/auth-store.test.ts`** — All 11 existing tests migrated to `store.update(...)` API, plus 7 new edge case tests: getter values after update, idempotent updates, mid-loading auth changes, concurrent `waitForAuth()` callers, cross-phase callers, and `invalidateRouter` assignability
- **`eslint.config.js`** — Route files (`src/routes/**/*.tsx`) exempt from `react-refresh/only-export-components` (TanStack Router convention); `buttonVariants` allowed as a co-export (shadcn/ui convention)
- **`convex/seed.ts`** — Suppressed `no-explicit-any` for the intentional `as any` cast on dynamic auth table iteration

### Formatting-only changes
All remaining file changes (UI components, routes, tests, docs, convex backend, e2e tests) are purely Prettier auto-formatting and Tailwind class sorting — no semantic changes.

Closes #57

## Test plan

- [x] `pnpm lint` passes with zero errors and zero warnings
- [x] `pnpm format:check` passes (all files formatted)
- [x] `pnpm vitest run` — all 180 tests pass (including 7 new auth-store edge case tests)
- [x] Pre-commit hook triggers lint-staged on `git commit`
- [ ] CI workflow passes on the PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,13 +82,30 @@
...
     "wrangler": "^4.61.0"
+  },
+  "lint-staged": {
+    "*.{js,ts,tsx}": [
+      "eslint --fix",
+      "prettier --write"
+    ],
+    "*.{json,md,yml,yaml,css}": [
+      "prettier --write"
+    ]
```
[the remark this answers is not in the document]
Comment (ciampo) [reply package.json]: `wrangler.jsonc` uses the `.jsonc` extension which signals JSONC format — trailing commas are valid and parsed correctly by wrangler's `jsonc-parser`. No need to exclude it from Prettier.
